lib/SearchClient/Exception/ClientException.php

```diff
@@ -0,0 +1,31 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\SearchClient\Exception;
+
+use RuntimeException;
+use Throwable;
+
+final class ClientException extends RuntimeException
+{
+    public function __construct(
+        mixed $message = '',
+        mixed $code = 0,
+        Throwable $previous = null
```

Comment:
Shouldn't it be like this? See: https://www.php.net/manual/de/exception.construct.php

```suggestion
        string $message = '',
        int $code = 0,
        ?Throwable $previous = null,
```

Reply:
Apart from that, this constructor does nothing more than pass the parameter to the parent element, so it is not needed at all.